source/io/module_javaScriptInvoke.js

```diff
@@ -131,156 +138,123 @@ var assignJavaScriptInvoke;
-                isValidReturnType: function (value) {
-                    return value instanceof Uint8Array;
-                }
-            },
-            ArrayUInt16: {
-                reader: Module.eggShell.dataReadTypedArray,
-                writer: Module.eggShell.dataWriteTypedArray,
-                isValidReturnType: function (value) {
-                    return value instanceof Uint16Array;
-                }
-            },
-            ArrayUInt32: {
-                reader: Module.eggShell.dataReadTypedArray,
-                writer: Module.eggShell.dataWriteTypedArray,
-                isValidReturnType: function (value) {
-                    return value instanceof Uint32Array;
-                }
-            },
-            ArraySingle: {
-                reader: Module.eggShell.dataReadTypedArray,
-                writer: Module.eggShell.dataWriteTypedArray,
-                isValidReturnType: function (value) {
-                    return value instanceof Float32Array;
-                }
-            },
-            ArrayDouble: {
-                reader: Module.eggShell.dataReadTypedArray,
-                writer: Module.eggShell.dataWriteTypedArray,
-                isValidReturnType: function (value) {
-                    return value instanceof Float64Array;
-                }
-            },
-            JavaScriptRefNum: {
-                reader: dataReadJavaScriptRefNum,
-                writer: dataWriteJavaScriptRefNum,
-                isValidReturnType: function () {
-                    return true;
+        var createJavaScriptInvokePeeker = function () {
+            var visitNumeric = function (valueRef) {
+                return Module.eggShell.readDouble(valueRef);
+            };
+
+            var visitNumeric64 = function (valueRef) {
+                return JSON.parse(Module.eggShell.readJSON(valueRef));
+            };
+
+            return {
+                visitInt8: visitNumeric,
+                visitInt16: visitNumeric,
+                visitInt32: visitNumeric,
+                visitInt64: visitNumeric64,
+                visitUInt8: visitNumeric,
+                visitUInt16: visitNumeric,
+                visitUInt32: visitNumeric,
+                visitUInt64: visitNumeric64,
+                visitSingle: visitNumeric,
+                visitDouble: visitNumeric,
+                visitEnum8: visitNumeric,
+                visitEnum16: visitNumeric,
+                visitEnum32: visitNumeric,
+
+                visitBoolean: function (valueRef) {
+                    return Module.eggShell.readDouble(valueRef) !== 0;
+                },
+
+                visitString: function (valueRef) {
+                    return Module.eggShell.readString(valueRef);
+                },
+
+                visitArray: function (valueRef) {
+                    return Module.eggShell.readTypedArray(valueRef);
+                },
+
+                visitJSObjectRefnum: function (valueRef) {
```

Comment:
Should remove visitJSObjectRefnum from this peeker and poker like discussed

Reply:
I am marking this resolved and keeping the one on peeker open. While resolving that, we should address this as well the same way anyway.